git_perf/src/config.rs

```diff
@@ -1,55 +1,129 @@
 use anyhow::Result;
+use config::{Config, ConfigError, File, FileFormat};
 use std::{
-    fs::File,
+    env,
+    fs::File as StdFile,
     io::{Read, Write},
+    path::{Path, PathBuf},
 };
 use toml_edit::{value, Document};
 
-use crate::git::git_interop::get_head_revision;
+use crate::git::git_interop::{get_head_revision, get_repository_root};
 
 // Import the CLI types for dispersion method
 use git_perf_cli_types::DispersionMethod;
 
+/// Get the main repository config path (always in repo root)
+fn get_main_config_path() -> PathBuf {
+    // Use git to find the repository root - hard fail if not found
+    let repo_root = get_repository_root()
+        .expect("Failed to determine repository root - must be run from within a git repository");
+
+    if repo_root.is_empty() {
+        panic!("Repository root is empty - must be run from within a git repository");
```

Comment:
@cursoragent we don't use panic. Use a proper error type and propagate it upwards.

Reply:
@cursoragent did you make changes for this?